source/math/FloatEuclidTransform.ooc

```diff
@@ -43,6 +46,22 @@ FloatEuclidTransform: cover {
 		this rotation == other rotation &&
 		this scaling == other scaling
 	}
+	convolveCenter: static func (euclidTransforms: VectorList<This>, kernel: FloatVectorList) -> This {
+		result := This new()
+		if (euclidTransforms count > 0) {
+			result scaling = 0.0f
+			quaternions := VectorList<Quaternion> new(euclidTransforms count)
+			for (i in 0 .. euclidTransforms count) {
+				result translation = result translation + euclidTransforms[i] translation * kernel[i]
```

Comment:
I'm tempted to optimize this (each `VectorList` access will result in a memcpy, and you're accessing the same elements 3 and 2 times in each iteration of the loop), but I guess it can wait until we see how much it actually affects performance...